runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -4102,8 +4137,132 @@ mod development {
 				});
 			}
 
+			fn increase_fulfill_decrease_fulfill_partial_increase<T: Runtime + FudgeSupport>() {
+				let mut env = FudgeEnv::<T>::from_parachain_storage(
+					Genesis::default()
+						.add(genesis::balances::<T>(cfg(1_000)))
+						.storage(),
+				);
+
+				setup_test(&mut env);
+
+				env.parachain_state_mut(|| {
+					let pool_id = POOL_ID;
+					let investor: AccountId = AccountConverter::<T, LocationToAccountId>::convert(
+						(DOMAIN_MOONBEAM, Keyring::Bob.into()),
+					);
+					let pool_currency: CurrencyId = AUSD_CURRENCY_ID;
+					let foreign_currency: CurrencyId = USDT_CURRENCY_ID;
+					let pool_currency_decimals = currency_decimals::AUSD;
+					let invest_amount_pool_denominated: u128 = 10 * decimals(18);
+					let trader: AccountId = Keyring::Alice.into();
+					create_currency_pool::<T>(
+						pool_id,
+						pool_currency,
+						pool_currency_decimals.into(),
+					);
+
+					// USDT investment preparations
+					let invest_amount_foreign_denominated = enable_usdt_trading::<T>(
+						pool_currency,
+						invest_amount_pool_denominated,
+						true,
+						true,
+						true,
+					);
+
+					// Do first investment and fulfill swap order
+					do_initial_increase_investment::<T>(
+						pool_id,
+						invest_amount_foreign_denominated,
+						investor.clone(),
+						foreign_currency,
+					);
+					let swap_order_id = pallet_foreign_investments::Swaps::<T>::swap_id_from(
+						&investor,
+						default_investment_id::<T>(),
+						pallet_foreign_investments::Action::Investment,
+					)
+					.expect("Swap order exists; qed");
+					fulfill_swap_into_pool::<T>(
+						pool_id,
+						swap_order_id,
+						invest_amount_pool_denominated,
+						invest_amount_foreign_denominated,
+						trader.clone(),
+					);
+
+					// Decrease pending pool swap by same amount
+					let decrease_msg_pool_swap_amount = LiquidityPoolMessage::DecreaseInvestOrder {
+						pool_id,
+						tranche_id: default_tranche_id::<T>(pool_id),
+						investor: investor.clone().into(),
+						currency: general_currency_index::<T>(foreign_currency),
+						amount: invest_amount_foreign_denominated,
+					};
+					assert_ok!(pallet_liquidity_pools::Pallet::<T>::submit(
+						DEFAULT_DOMAIN_ADDRESS_MOONBEAM,
+						decrease_msg_pool_swap_amount
+					));
+
+					// Fulfill decrease swap partially
+					let swap_order_id = pallet_foreign_investments::Swaps::<T>::swap_id_from(
+						&investor,
+						default_investment_id::<T>(),
+						pallet_foreign_investments::Action::Investment,
+					)
+					.expect("Swap order exists; qed");
+					assert_ok!(pallet_order_book::Pallet::<T>::fill_order(
+						RawOrigin::Signed(trader.clone()).into(),
+						swap_order_id,
+						invest_amount_pool_denominated / 4 * 3
```

Comment:
I think this is wrong. The multiplication has priority. So it is: `AMOUNT / 12`

Reply:
It should not, rust math is distributive such that `x * y  / z = (x * y) / z`. Safety first, I checked by doing `3 * invest_amount / 4` and it didn't change anything.

Reply:
I think you are mixing it up with the order of operations for addition/subtraction and multiplication/division: `a + b + c = a + (b * c)`

Reply:
Yeah, in the same way `*` takes more priority over `+`, I thought it was for  `*` and `/`. Thanks for enlighten me 🙌🏻